File: codeforces/cf71A_way_too_long_words.hs
-- Vicfred
-- https://codeforces.com/problemset/problem/71/A
-- implementation
transform :: String -> String
transform s
    | (length s <= 10) = s
    | otherwise = ([head s] ++ show (length s - 2) ++ [last s])

main :: IO ()
main =  do
    w <- getLine
    words <- getContents
    putStrLn . unlines . map transform . lines $ words

File: codeforces/cf71A_way_too_long_words___.cpp
// https://codeforces.com/problemset/problem/71/A
#include<iostream>
#include<string>

int solve(std::string word) {
    return (word.end() - 1) - (word.begin() + 1);
}

int main() {
    int n;
    std::cin >> n;
    std::string current;
    while (n > 0) {
        std::cin >> current;
        if (current.length() > 10) {
            std::cout << current.at(0) << solve(current) << current.back() << std::endl;
        } else {
            std::cout << current << std::endl;
        }
        n--;
    }
    return 0;
}
